src/wechaty/user/contact.py

```diff
@@ -132,25 +131,49 @@ async def find_all(cls: Type[Contact],
         log.info('find_all() <%s, %s>', cls, query)
 
         contact_ids = await cls.get_puppet().contact_list()
+        # filter Contact by contact id to make sure its valid if contact_id.startswith('wxid_')
         contacts = [Contact.load(contact_id) for contact_id in contact_ids]
 
         # load contact parallel using asyncio.gather method
-        await asyncio.gather(*[contact.ready() for contact in contacts])
 
         # async load
         batch_size = 16
-        # slice contacts by batch_size
-        contacts = contacts[:batch_size * (len(contacts) // batch_size)]
-
+        batch_index = 0
         contact_result_list: List[Contact] = []
-
-        for contact in contacts:
+        # slice contacts by batch_size
+        while batch_index * batch_size < len(contacts):
+            batch_contactList = contacts[batch_index * batch_size: (batch_index + 1) * batch_size]
             try:
-                await contact.ready()
-                contact_result_list.append(contact)
+                await asyncio.gather(*[contact.ready() for contact in batch_contactList])
+                contact_result_list += batch_contactList
             except RuntimeError as exception:
                 log.info('load contact occur exception: %s', exception.args)
+            batch_index = batch_index + 1
+
+        if isinstance(query, str):
+            contact_result_list = list(
+                filter(
+                    lambda x: False if not x.payload else
+                    (x.payload.alias.__contains__(query)) or
+                    (x.payload.id.__contains__(query)) or
+                    (x.payload.name.__contains__(query)) or
+                    (x.payload.weixin.__contains__(query)),
+                    contact_result_list
```

Comment:
`id` and `weixin` should be `equal` operation, `alias`, `name` and the `topic` below should be `contains` operation. 

How do you think about it ?

Reply:
`id` and `weixin` are not equal if `id` starts with `wxid_`, and I find if a Conact is a friend, then you can get both `wxid_` started id and  normal wechat ID.

Reply:
Ok, It fine to work at this way.